src/doublyLinkedList.ts

```diff
@@ -0,0 +1,83 @@
+// found this here - https://www.cnblogs.com/answer1215/p/7623103.html
+
+/**
+ * Linked list node
+ */
+export interface DoublyLinkedListNode<T> {
+	value: T;
+	next?: DoublyLinkedListNode<T>;
+	prev?: DoublyLinkedListNode<T>;
+}
+
+/**
+ * Linked list for items of type T
+ */
+export class DoublyLinkedList<T> {
+	public head?: DoublyLinkedListNode<T> = undefined;
+	public tail?: DoublyLinkedListNode<T> = undefined;
+
+	/**
+	* Adds an item in O(1)
+	**/
+	add(value: T) {
+		const node: DoublyLinkedListNode<T> = {
+			value,
+			next: undefined,
+			prev: undefined,
+		};
+
+		if (!this.head) {
+			this.head = node;
+		}
+		if (this.tail) {
+			this.tail.next = node;
+			node.prev = this.tail;
+		}
+		this.tail = node;
+	}
+  
+	/**
+	* FIFO removal in O(1)
+	*/
+	dequeue(): T | undefined {
+		if (this.head) {
+			const value = this.head.value;
+			this.head = this.head.next;
+			if (!this.head) {
+				this.tail = undefined;
```

Comment:
there is something wrong in this function with indenting and also the `else` being on a new line